### PR TITLE
Add gfx940 target for f32/f64 types

### PR DESCRIPTION
**Pull request by mkarunan**
Add gfx940 to the targets
Enable the existing datatypes support on the above target

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,9 +2,9 @@
 hiptensor is AMD's C++ library for accelerating tensor primitives based on the composable kernel library, through general purpose kernel languages, like HIP C++.
 
 ## GPU Support
-* AMD CDNA class GPU featuring matrix core support: gfx908, gfx90a as 'gfx9'
+* AMD CDNA class GPU featuring matrix core support: gfx908, gfx90a as 'gfx9', gfx940
```
Comment (dlangbe) [inline README.md]: Should read:
AMD CDNA class GPU featuring matrix core support: gfx908, gfx90a, gfx940 as 'gfx9'

**Review by dlangbe — CHANGES_REQUESTED** [review]

File: samples/01_contraction/simple_bilinear_contraction_f32.cpp [hunk 2]
```diff
@@ -58,7 +58,8 @@ inline bool isF32Supported()
     std::string deviceName(mProps.gcnArchName);
 
     return (deviceName.find("gfx908") != std::string::npos)
-           || (deviceName.find("gfx90a") != std::string::npos);
+            || (deviceName.find("gfx90a") != std::string::npos)
```
Comment (cgmillette) [inline samples/01_contraction/simple_bilinear_contraction_f32.cpp]: Can remove duplicate code and replace with the common.hpp version

File: CMakeLists.txt [hunk 3]
```diff
@@ -79,7 +79,7 @@ include(ROCMInstallSymlinks)
 include(CheckCXXCompilerFlag)
 
 rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
-  TARGETS "gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+" )
+  TARGETS "gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx940:xnack-;gfx940:xnack+" )
```
Comment (cgmillette) [inline CMakeLists.txt]: @mkarunan for the targets, let's just set gfx940 without xnack. It will default to xnack:any which is sufficient for now

**Review by cgmillette — APPROVED** [review]
LGTM

**Pull request merged**